test/producers/application_producer_test.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class ApplicationProducerTest < ActiveSupport::TestCase
+  class MockProducer < ApplicationProducer
+    BROKERS = ['broker1'].freeze
+  end
+
+  test 'handles SSL settings' do
+    MockProducer::SECURITY_PROTOCOL = 'ssl'
+    MockProducer::SSL_CA_LOCATION = '/app/test/fixtures/files/test_ca.crt'
+
+    config = {
+      client_id: ApplicationProducer::CLIENT_ID,
+      ssl_ca_cert: "very secure\n"
+    }
+
+    assert_equal config, MockProducer.send(:kafka_config)
```

Comment:
I'm not usually a fan of testing private methods, but this is easier than trying to mock out `Kafka`.